src/lib.rs

```diff
@@ -260,6 +274,81 @@ impl Pixels {
     pub fn get_frame(&mut self) -> &mut [u8] {
         &mut self.pixels
     }
+
+    /// Calculate the pixel location from a physical location on the window,
+    /// dealing with window resizing, scaling, and margins. Takes a physical
+    /// position (x, y) within the window, and returns a pixel position (x, y).
+    ///
+    /// The location must be given in physical units (for example, winit's `PhysicalLocation`)
+    ///
+    /// If the given physical position is outside of the drawing area, this
+    /// function returns an `Err` value with the pixel coordinates outside of
+    /// the screen, using isize instead of usize.
+    ///
+    /// ```no_run
+    /// # use pixels::Pixels;
+    /// # let surface = wgpu::Surface::create(&pixels_mocks::RWH);
+    /// # let surface_texture = pixels::SurfaceTexture::new(1024, 768, surface);
+    /// const WIDTH:  u32 = 320;
+    /// const HEIGHT: u32 = 240;
+    ///
+    /// let mut pixels = Pixels::new(WIDTH, HEIGHT, surface_texture)?;
+    ///
+    /// // A cursor position in physical units
+    /// let cursor_position: (f64, f64) = winit::dpi::PhysicalPosition::new(0, 0).into();
+    ///
+    /// // Convert it to a pixel location
+    /// let pixel_position: (usize, usize) = pixels.window_pos_to_pixel(cursor_position)
+    ///     .unwrap_or_else(|(x, y)| ( // Clamp the output to within the screen
+    ///         x.max(0).min(WIDTH as isize - 1) as usize,
+    ///         y.max(0).min(HEIGHT as isize - 1) as usize,
+    ///     ));
+    /// # Ok::<(), pixels::Error>(())
+    /// ```
+    pub fn window_pos_to_pixel(
+        &self,
+        physical_position: (f64, f64),
+    ) -> Result<(usize, usize), (isize, isize)> {
+        let physical_size = (
+            self.surface_texture.width as f32,
+            self.surface_texture.height as f32,
+        );
+
+        let pixels_width = self.texture_extent.width as f32;
+        let pixels_height = self.texture_extent.height as f32;
+
+        let pos = [
+            (physical_position.0 as f32 / physical_size.0 as f32 - 0.5) * pixels_width,
+            (physical_position.1 as f32 / physical_size.1 as f32 - 0.5) * pixels_height,
+        ];
+        // Pos vector: (x, y, 0, 1)
+
+        let matrix = self.scaling_matrix_inverse;
+
+        // new_pos: (x, y, w) - z isn't needed so it isn't calculated
+        let new_pos = [
+            matrix[0][0] * pos[0] + matrix[0][1] * pos[1] + matrix[0][3],
+            matrix[1][0] * pos[0] + matrix[1][1] * pos[1] + matrix[1][3],
+            matrix[3][0] * pos[0] + matrix[3][1] * pos[1] + matrix[3][3],
```

Comment:
I looked around at the options (`ultraviolet`, `cgmath`, and `vek`), and I would personally prefer using `ultraviolet`, or `cgmath` as a second option.  My problem with `vek` is with compile times.  For a clean release build on my computer, `ultraviolet` takes 2s and `cgmath` takes 6s, while `vek` takes 30 seconds to compile.  As far as I can tell, there is no way of avoiding having it pull in `serde` without using the `no_std` version.  `Ultraviolet` seems like the best option here, as it supports using `bytemuck` to convert it to a byte slice. 

Ultimately it's your decision, as you will likely have to use the chosen library for the rest of `pixels`.  `vek` seems like it would add a large number of dependencies and compile time for features that `pixels` doesn't use, so I would personally prefer a smaller library.